File: H6_ChicBotique/Database/Entities/Order.cs
using H5_Webshop.Database.Entities;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace H6_ChicBotique.Database.Entities
{
    public class Order
    {
        [Key]
        public int Id { get; set; }

        [Column(TypeName = "Date")]  //this is a columnAttribute from System.CoponentModel.DataAnnotations (defined in enityframework.dll)
        public DateTime OrderDate { get; set; }

        [ForeignKey("AccountId")]
        public Guid AccountId { get; set; }
        public AccountInfo AccountInfo { get; set; }
        public ShippingDetails ShippingDetails { get; set; }

        public List<OrderDetails> OrderDetails { get; set; } = new();

        [ForeignKey("PaymentId")]
        public int PaymentId { get; set; }
        public Payment Payment { get; set; }
    }

}

File: H6_ChicBotiqueClientSide/src/app/_models/order.ts

import { CartItem } from "./cartItem";
import { ShippingDetails } from "./shippingdetails";


export interface Order {

    id?: number;
    accountId: string;
    orderDate?: Date;
    amount: number;
    transactionId: string;
    status: string;
    paymentMethod:string;
    timePaid?:Date
    shippingDetails: ShippingDetails;

    orderDetails: CartItem[];


}
